### PR TITLE
[TE] Extend autotune to take user configurable parameters

### PR DESCRIPTION
**Pull request by cecilynie**
Cloned from https://github.com/linkedin/pinot/pull/1736
Previously auto-tuning is taking fixed parameters. Now extend auto-tuning to take free-formed configuration and complete a two-sided tuning.

### CODE REVIEW

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomalydetection/alertFilterAutotune/AlertFilterAutoTune.java [hunk 1]
```diff
@@ -1,26 +1,14 @@
...
-  Map<String, String> initiateAutoTune(List<MergedAnomalyResultDTO> anomalyResults, int nExpectedAnomalies) throws Exception;
-
-  // Tune the alert filter properties using labeled anomalies
-  // precision and recall need to be improved over old setting
-  Map<String, String> tuneAlertFilter(List<MergedAnomalyResultDTO> anomalyResults, double currentPrecision, double currentRecall) throws Exception;
-
-  // True if model has been updated (improved)
-  boolean isUpdated();
+  // Tune alert filter based on training data
+  Map<String, String> tuneAlertFilter() throws Exception;
```
Comment (bl44) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomalydetection/alertFilterAutotune/AlertFilterAutoTune.java]: please provide proper comments (java doc) for this interface

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/datalayer/dto/AutotuneConfigDTO.java [hunk 2]
```diff
@@ -1,11 +1,53 @@
...
 
 
+/**
+ * Auto Tune Configuration DTO. Given Alert filter, add alert filter setting to autotune config properties
+ * When autotune being triggered, autotuneConfig will be passed to tuning steps
+ */
 public class AutotuneConfigDTO extends AutotuneConfigBean {
   private static final Logger LOGGER = LoggerFactory.getLogger(AutotuneConfigDTO.class);
+  private AlertFilter alertFilter = new DummyAlertFilter();
+  private Properties tuningProps = new Properties();
```
Comment (bl44) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/datalayer/dto/AutotuneConfigDTO.java]: should this be a field in AutotuneConfigBean?

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/datalayer/pojo/AutotuneConfigBean.java [hunk 3]
```diff
@@ -26,7 +27,7 @@
   // The goal of the autotune
   private double goal;
   // The tuned properties configuration
-  private Map<String, String> configuration;
+  private Map<String, String> configuration = new HashMap<>();
```
Comment (bl44) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/datalayer/pojo/AutotuneConfigBean.java]: Is this change truly necessary? There was no problem without this default initialization.

**Comment by bl44** [comment]
Since this change will modify database tables, please wrote proper instrunctions for product release on thirdeye, they may facing errors without knowing this particular change.

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomalydetection/alertFilterAutotune/AlertFilterAutoTune.java (hunk 1 above)
Comment (cecilynie) [reply thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomalydetection/alertFilterAutotune/AlertFilterAutoTune.java]: Will do.

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/datalayer/dto/AutotuneConfigDTO.java (hunk 2 above)
Comment (cecilynie) [reply thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/datalayer/dto/AutotuneConfigDTO.java]: It's to be decided. AutotuneConfigBean contains entities to be wrote to db. Not sure if these tuning related properties need to put to bean.

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/datalayer/pojo/AutotuneConfigBean.java (hunk 3 above)
Comment (cecilynie) [reply thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/datalayer/pojo/AutotuneConfigBean.java]: It is needed in Anomaly Detection Library.

**Pull request merged**